polkadot/node/subsystem-util/src/runtime/mod.rs

```diff
@@ -507,3 +507,32 @@ pub async fn request_min_backing_votes(
 		min_backing_votes_res
 	}
 }
+
+/// Request the node features enabled in the runtime.
+/// Pass in the session index for caching purposes, as it should only change on session boundaries.
+/// Prior to runtime API version 9, just return `None`.
+pub async fn request_node_features(
+	parent: Hash,
+	session_index: SessionIndex,
+	sender: &mut impl overseer::SubsystemSender<RuntimeApiMessage>,
+) -> Result<Option<NodeFeatures>> {
+	let res = recv_runtime(
+		request_from_runtime(parent, sender, |tx| {
+			RuntimeApiRequest::NodeFeatures(session_index, tx)
+		})
+		.await,
+	)
+	.await;
+
```

Comment:
On session changes it would be a good idea to emit an warning to upgrade the node if not all bits make sense(at least a required feature we don't have)

Reply:
I don't think that's doable if we want to be able to add new features without a runtime upgrade.
The runtime does not have any idea of what the feature bits mean. They only make sense on the node-side.

If we want to emit warnings for new features that are added, they have to come via a runtime upgrade (which would defeat the purpose of what I've done with this PR so far in order to have seamless feature addition/enabling). I think it's best to leave it as it is

Reply:
>The runtime does not have any idea of what the feature bits mean. They only make sense on the node-side.
> 

Yes exactly, those are bits that only the client understands. This is just to inform the operator, that currently a client requirement is not fulfilled and a node upgrade is required.

> If we want to emit warnings for new features that are added, they have to come via a runtime upgrade (which would defeat the purpose of what I've done with this PR so far in order to have seamless feature addition/enabling). I think it's best to leave it as it is

No runtime upgrade is needed as the client will see a new `1` bit  when the feature gets enabled. Even if it doesn't know what it means it can still print the warning saying to upgrade the node.

Reply:
makes sense, I didn't fully get this from the beginning.
I can make the `RuntimeApiSubsystem` track the last session and when seeing a new session check the highest bit set returned from the `node_features` runtime API. If that's higher than some `LAST_NODE_FEATURE` constant on the node side, issue a warning.

As spoken on element, I'll add this on the next PR that uses this runtime API, as it'd be redundant right now.
Good suggestion 👍🏻